### DISCUSSION TITLE
How to limit the map so that the actor cannot go beyond its boundaries

### DISCUSSION THREAD

**Discussion by Evgenii190** [excaliburjs/Excalibur#2097] [Q&A]
I read the documentation, but I could not find how to do it. Question: How to limit the map so that the actor cannot go beyond its boundaries

**Comment by kamranayub** [comment]
Hey! This would be logic you'd add to the actor, since there could be many different actors with all sorts of logic.

The way to accomplish it is by using the `onPostUpdate` hook and checking the [world bounds (`Engine.getWorldBounds()`)](https://excaliburjs.com/docs/api/edge/classes/Engine.html#getWorldBounds) vs. the actor's current position. There is an example of this in the Getting Started game with the `Ball` actor (it checks `Engine.drawWidth` which is the canvas draw width):

https://excaliburjs.com/docs/getting-started#creating-the-ball-with-actors
